src/antsibull/dep_closure.py

```diff
@@ -0,0 +1,85 @@
+# coding: utf-8
+# Author: Toshio Kuratomi <tkuratom@redhat.com>
+# License: GPLv3+
+# Copyright: Ansible Project, 2021
+"""Check collection dependencies."""
+
+import json
+import pathlib
+
+from collections import namedtuple
+from typing import Dict, List, Mapping
+
+from semantic_version import Version as SemVer, SimpleSpec as SemVerSpec
+
+from antsibull_core import app_context
+
+
+CollectionRecord = namedtuple('CollectionRecord', ('version', 'dependencies'))
+
+
+def parse_manifest(collection_dir: pathlib.Path) -> Mapping[str, CollectionRecord]:
+    '''Parse MANIFEST.json for a collection.'''
+    manifest = collection_dir.joinpath('MANIFEST.json')
+    with manifest.open() as f:
+        manifest_data = json.load(f)['collection_info']
+
+    collection_record = {
+        f'{manifest_data["namespace"]}.{manifest_data["name"]}':
+            CollectionRecord(manifest_data['version'], manifest_data['dependencies'])
+    }
+
+    return collection_record
+
+
+def analyze_deps(collections: Mapping[str, CollectionRecord]) -> List[str]:
+    '''Analyze dependencies of a set of collections. Return list of errors found.'''
+    errors = []
+
+    # Look at dependencies
+    # make sure their dependencies are found
+    for collection_name, collection_info in collections.items():
+        for dep_name, dep_version_spec in collection_info.dependencies.items():
+            if dep_name not in collections:
+                errors.append(f'{collection_name} missing: {dep_name} ({dep_version_spec})')
+                continue
+
+            dependency_version = SemVer(collections[dep_name].version)
+            if dependency_version not in SemVerSpec(dep_version_spec):
+                errors.append(f'{collection_name} version_conflict:'
+                              f' {dep_name}-{str(dependency_version)} but needs'
+                              f' {dep_version_spec}')
```

Comment:
This produces: ``community.okd version_conflict: kubernetes.core-2.3.0 but needs >=2.1.0,<2.3.0`` which isn't bad but I could see it get lost in the noise during a build. Maybe we could prepend it with ``ERROR: `` or something like that ? Similar thoughts in other places where we print them.

Reply:
Good idea, I implemented that in 88dd4b2cc544effc78ade1492cf953246e968add.

The other place where they are printed is the `validate-deps` subcommand, where (similar to other linting subcommands in antsibull-changelog and antsibull-docs) stdout output is only generated if something is broken. There's no need to prepend things, this will only make it harder to process the output in other tools.